File: Ser2023/Company/Company.Web/Modules/Company/BusinessEntityAddress/BusinessEntityAddressColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Company.Company.Columns;

[ColumnsScript("Company.BusinessEntityAddress")]
[BasedOnRow(typeof(BusinessEntityAddressRow), CheckNames = false)]
public class BusinessEntityAddressColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public long BusinessEntityId { get; set; }
    public long AddressId {  get; set; }
    //public string AddressLine1 { get; set; }
    public long AddressTypeId { get; set; }
    //public string AddressTypeName {  get; set; }
    //public Guid Rowguid { get; set; }
}

File: 2025/Company/Company.Web/Modules/ServerTypes/Company/BusinessEntityAddressColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BusinessEntityAddressRow } from "./BusinessEntityAddressRow";

export interface BusinessEntityAddressColumns {
    BusinessEntityId: Column<BusinessEntityAddressRow>;
    AddressLine1: Column<BusinessEntityAddressRow>;
    AddressTypeName: Column<BusinessEntityAddressRow>;
}

export class BusinessEntityAddressColumns extends ColumnsBase<BusinessEntityAddressRow> {
    static readonly columnsKey = 'Company.BusinessEntityAddress';
    static readonly Fields = fieldsProxy<BusinessEntityAddressColumns>();
}